### PR TITLE
BAH-1670, BAH-1796 - Dockerized and refactored Hip-atomfeed-listener

### PR DESCRIPTION
**Pull request by SanoferSameera**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+version: "3.7"
+services:
+  hip-atomfeed-listener:
+    container_name: hip-atomfeed-listener
+    image: bahmniindiadistro/hip-atomfeed
+    environment:
+      - PGPASSWORD=welcome
+      - DATABASE_URL=jdbc:postgresql://db:5432/
+      - DATABASE_NAME=bahmni_pacs
```
Comment (swatigogia2020) [inline docker-compose.yml:9]: need to refactor name

File: hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/atomfeed/client/Constants.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.bahmni.module.hipfeedintegration.atomfeed.client;
+
+public class Constants {
+
+    public static final String PACS_CONCEPT_SOURCE_NAME = "PACS Procedure Code";
```
Comment (swatigogia2020) [inline hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/atomfeed/client/Constants.java:5]: are we using this string?

File: hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/HipService.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.io.UnsupportedEncodingException;
+
+import static org.bahmni.module.hipfeedintegration.services.OpenMRSService.getURLPrefix;
+
+@Component
+public class HipService {
+
+    private static final Logger logger = LoggerFactory.getLogger(EncounterFeedWorker.class);
+
+    String newCareContext = "/hiprovider/v0.5/hip/new-carecontext";
```
Comment (swatigogia2020) [inline hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/HipService.java:23]: we can move this to constant file

File: hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/HipService.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+
+
+    private HttpPost createRequest(String uri, String requestBody) throws UnsupportedEncodingException {
+        HttpPost httpPost = new HttpPost(uri);
+        StringEntity entity = new StringEntity(requestBody, "application/json", "UTF-8");
+        httpPost.setEntity(entity);
+        httpPost.setHeader("Content-Type", "application/json");
+        httpPost.setHeader("CORRELATION_ID", null);
+        logger.warn("===========================request begin================================================");
```
Comment (swatigogia2020) [inline hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/HipService.java:56]: Once tested remove the warning log, maybe have a short meaningful debug log

File: hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/OpenMRSService.java [hunk 5]
```diff
@@ -1,29 +1,32 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 @Component
 public class OpenMRSService {
 
     String patientRestUrl = "/openmrs/ws/rest/v1/patient/";
     String newCareContextUrl = "/openmrs/ws/rest/v1/hip/careContext/new?patientUuid=";
+    String globalPropertyurl = "/openmrs/ws/rest/v1/systemsetting/";
```
Comment (swatigogia2020) [inline hip-atomfeed/src/main/java/org/bahmni/module/hipfeedintegration/services/OpenMRSService.java:29]: can we move these APIs to constant file

**Review by swatigogia2020 — CHANGES_REQUESTED** [review]

**Review by N0-man — APPROVED** [review]

**Pull request merged**